### PR TITLE
Add empty copy checks

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Waiting on approval of [internal Vulkan Change](https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5105) first... hence the `todo` in the VUIDs

edit: I guess there was a `LogWarning` for `kVUID_Core_Image_ZeroAreaSubregion` already... will update

edit: working group approved change, just waiting on new spec to rollout to get VUID and will update and can roll this in with 1.3.209 header update

**update:** has new VUIDs ready to go

### CODE REVIEW

File: tests/vklayertests_command.cpp [hunk 1]
```diff
@@ -3500,6 +3517,99 @@ TEST_F(VkLayerTest, CopyImageDstSizeExceeded) {
     m_commandBuffer->end();
 }
 
+TEST_F(VkLayerTest, CopyImageZeroSize) {
+    ASSERT_NO_FATAL_FAILURE(Init());
```
Comment (ziga-lunarg) [inline tests/vklayertests_command.cpp:3521]: Could use test description

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -3029,6 +3029,29 @@ bool CoreChecks::ValidateCmdCopyImage(VkCommandBuffer commandBuffer, VkImage src
                 func_name, i, region.dstSubresource.aspectMask);
         }
 
+        // Make sure not a empty region
+        if (src_copy_extent.width == 0) {
+            vuid = is_2 ? "VUID-VkImageCopy2-extent-06668" : "VUID-VkImageCopy-extent-06668";
+            skip |=
+                LogError(command_buffer, vuid,
+                         "%s: pRegion[%" PRIu32 "] extent.width (%" PRIu32 ") must not be zero as empty copies are not allowed.",
+                         func_name, i, src_copy_extent.width);
```
Comment (ziga-lunarg) [inline layers/buffer_validation.cpp:3038]: `src_copy_extent.width` is 0 so it could just be written as 0 instead of parameter

**Review by ziga-lunarg — APPROVED** [review]

File: layers/buffer_validation.cpp (hunk 2 above)
Comment (sfricke-samsung) [reply layers/buffer_validation.cpp]: good call

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

**Pull request merged**